### PR TITLE
Share dialog

### PR DESCRIPTION
**Pull request by Georgehe4**
Share dialog now allows users to see summaries.

screen shot 2015-06-03 at 1 28 43 pm

### CODE REVIEW

**Comment by Georgehe4** [comment]
@devoncarew

File: lib/dialogs.dart [hunk 1]
```diff
@@ -72,18 +72,21 @@ class SharingDialog extends DDialog {
   final GistController gistController;
 
   ParagraphElement _text;
+  TextAreaElement _dialog;
```
Comment (devoncarew) [inline lib/dialogs.dart]: `_dialog`  ==> `summaryArea`?

File: lib/dialogs.dart [hunk 2]
```diff
@@ -72,18 +72,21 @@ class SharingDialog extends DDialog {
...
   DButton _cancelButton;
   DButton _shareButton;
   DButton _closeButton;
   DElement _div;
   DInput _padUrl;
   DInput _gistUrl;
+  String _summary;
 
-  SharingDialog(this.gistContainer, this.gistController) : super(title: 'Sharing') {
+  SharingDialog(this.gistContainer, this.gistController, {summary: ""}) : super(title: 'Sharing') {
```
Comment (devoncarew) [inline lib/dialogs.dart]: can probably change to `{this.summary}`

File: lib/dialogs.dart [hunk 3]
```diff
@@ -72,18 +72,21 @@ class SharingDialog extends DDialog {
...
     element.classes.toggle('sharing-dialog', true);
-
+    _summary = summary;
```
Comment (devoncarew) [inline lib/dialogs.dart]: and remove this line

File: lib/dialogs.dart [hunk 4]
```diff
@@ -109,6 +112,11 @@ class SharingDialog extends DDialog {
     _gistUrl.onClick.listen((_) => _gistUrl.selectAll());
   }
 
+  void updateSummary(String summary) {
```
Comment (devoncarew) [inline lib/dialogs.dart]: Let's call this something else, to indicate that it is also makes the dialog visible. `showWithSummary()`? `showSharingDialog(String summary)`?

File: lib/dialogs.dart [hunk 5]
```diff
@@ -130,7 +138,10 @@ class SharingDialog extends DDialog {
       // Show 'about to share'.
       _text.text = 'Sharing this pad will create a permanent, publicly visible '
           'copy on gist.github.com.';
-
+      _dialog.defaultValue = _summary;
```
Comment (devoncarew) [inline lib/dialogs.dart]: Maybe `= _summary == null ? '' : _summary` to handle nulls.

File: lib/dialogs.dart [hunk 6]
```diff
@@ -130,7 +138,10 @@ class SharingDialog extends DDialog {
...
+      _dialog.cols = 75;
+      _dialog.className = 'dialogBox';
```
Comment (devoncarew) [inline lib/dialogs.dart]: `'dialogBox'` ==> `'sharingSummaryText'`?

File: web/index.css [hunk 7]
```diff
@@ -54,6 +54,16 @@ header {
...
 }
 
+.dialogBox{
+  background-color: #333;
+  color: #c6c6c6;
+  font-family: 'Roboto', sans-serif;
+  font-size: 12px;
+  height: 70px;
+  width: 95%;
+  resize: none;
```
Comment (devoncarew) [inline web/index.css]: What does the text area look like w/o a border? Or with a more subtle border (`#555`)?

**Comment by devoncarew** [comment]
done with the review - mostly some comments on naming and some style comments for the dialog

**Pull request merged**